### PR TITLE
Fixed collection view layout for landscape (implements #45)

### PR DESCRIPTION
**Pull request by WERUreo**
Added a collection view flow layout for landscape orientation, making the cells span three across.  Also added code in the PetListingViewController to set the collection view layout to the appropriate layout when the orientation of the device changes.

### CODE REVIEW

File: PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift [hunk 1]
```diff
@@ -47,28 +47,47 @@ class PetListingViewController: UIViewController
...
     }
 
     ////////////////////////////////////////////////////////////
+
+    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator)
+    {
+        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
+
+        let currentOrientation = UIDevice.currentDevice().orientation
+        let isLandscape = (currentOrientation == .LandscapeLeft) || (currentOrientation == .LandscapeRight)
```
Comment (LondonAtlas) [inline PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift]: consider using:
func UIDeviceOrientationIsLandscape(_ orientation: UIDeviceOrientation) -> Bool

File: PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift [hunk 2]
```diff
@@ -47,28 +47,47 @@ class PetListingViewController: UIViewController
...
 
+    ////////////////////////////////////////////////////////////
+
     override func viewWillAppear(animated: Bool)
     {
         super.viewWillAppear(animated)
         self.navigationItem.title = NSLocalizedString("Town Of Lady Lake", comment: "")
     }
 
+    ////////////////////////////////////////////////////////////
```
Comment (WERUreo) [inline PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift]: Yup

File: PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import UIKit
+
+class HomeLandscapeCollectionViewLayout: UICollectionViewFlowLayout
+{
+    ////////////////////////////////////////////////////////////
+    // MARK: - Properties
+    ////////////////////////////////////////////////////////////
+
+    override var itemSize: CGSize
```
Comment (LondonAtlas) [inline PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift]: if itemSize is never set, change this to itemSize: CGSize {
return size
}

File: PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    ////////////////////////////////////////////////////////////
+    // MARK: - Initializers
+    ////////////////////////////////////////////////////////////
+
+    override init()
+    {
+        super.init()
+        setupLayout()
```
Comment (LondonAtlas) [inline PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift]: always use self when calling a function.

File: PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        super.init(coder: aDecoder)
+        setupLayout()
+    }
+
+    ////////////////////////////////////////////////////////////
+
+    func setupLayout()
+    {
+        minimumInteritemSpacing = 0
```
Comment (LondonAtlas) [inline PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift]: always use self

File: PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift (hunk 3 above)
Comment (WERUreo) [inline PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift]: What about the getter?

Comment (WERUreo) [inline PetAdoption-iOS/PetAdoption-iOS/Views/HomeLandscapeCollectionViewLayout.swift]: So, if I remove the setter, I get the following error:
`Getter for 'itemSize' with Objective-C selector 'itemSize' conflicts with getter for 'itemSize' from superclass 'UICollectionViewFlowLayout' with the same Objective-C selector.`

If I take everything in the getter, and just place that in the trailing closure for the property, I get this error:
`Cannot override mutable property with read-only property 'itemSize'`

I'm not sure what to do about each of those errors.  I've seen that code used in other projects and tutorials online.  I will admit that I'm unsure of exactly how that works.  If you've got some insight, I'm all ears. :)

File: PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift (hunk 2 above)
Comment (LondonAtlas) [inline PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift]: perhaps it is a personal preference but wouldn't be cleaner to only have them between sections?

Comment (WERUreo) [inline PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift]: I dunno, this is the way I code.  12 years of coding standards at my last job.  In lieu of a coding standard of our own, and given that at one time, no one else was developing on this project, I adopted my personal coding standard.  If my coding style really bothers everyone, I suppose I can change it...

Comment (LondonAtlas) [inline PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift]: all good, this is your project.

**Pull request merged**